enzy_htp/preparation/reactive_docking.py

```diff
@@ -0,0 +1,1285 @@
+"""Driver for the reactive docking functionality available in enzy_htp. The only function that that should be 
+called is dock_reactants(). All others are implementation functions that SHOULD NOT be used. 
+
+Author: Chris Jurich <chris.jurich@vanderbilt.edu>
+Date: 2023-07-28
+"""
+import os
+from pathlib import Path
+from collections import defaultdict
+from typing import List, Tuple, Set
+from copy import deepcopy
+
+import numpy as np
+import pandas as pd
+
+from enzy_htp import interface, config, _LOGGER, binding_energy
+from enzy_htp._interface.rosetta_interface import RosettaCst
+
+import enzy_htp.chemical as chem
+from enzy_htp import mutation as mm
+from enzy_htp.structure import PDBParser, Structure
+from enzy_htp.core import file_system as fs
+
+#TODO(CJ): I should go through and use existing values if cache mode is selected
+#TODO(CJ): need to be able to specify the size of the grid
+
+
+def dock_reactants(structure: Structure,
+                   reactants: List[str],
+                   constraints: List[RosettaCst] = None,
```

Comment:
Is it possible to decouple this input from Rosetta? e.g.: create a general constrain class that can be convert to the format of different software inside their own interfaces

Reply:
sure!